src/shared/common/Button.js

```diff
@@ -0,0 +1,83 @@
+// @flow
+import * as React from 'react';
+
+import { Component } from 'react';
+import { Text, View, StyleSheet, Pressable } from 'react-native';
+
+// loosely adapted from https://github.com/APSL/react-native-button
+const styles = StyleSheet.create({
+    button: {
+        height: 44,
+        flexDirection: 'row',
+        alignItems: 'center',
+        borderWidth: 1,
+        borderRadius: 8,
+        marginBottom: 10,
+        alignSelf: 'stretch',
+        justifyContent: 'center',
+    },
+    text: {
+        fontSize: 16,
+        lineHeight: 21,
+        fontWeight: 'bold',
+        letterSpacing: 0.25,
+        color: 'white',
+    },
+    opacity: {
+        opacity: 0.5,
+    },
+});
+
+type Props = {
+    children: any,
+    onPress: () => any,
+    textStyle: Object,
+    style: Object,
+    testID?: string,
+    isDisabled?: boolean,
+};
+
+class Button extends Component<Props> {
```

Comment:
If this is going to be a reusable component, maybe add a short docstring explaining the props that can be set on the button? Or maybe just add comments in the `type Props` definition above?